test/promql/tests/data_plane/latency_snapshot_to_static_integration_plr_creation_test.yaml

```diff
@@ -0,0 +1,99 @@
+rule_files:
+  - 'prometheus.latency_snapshot_to_static_integration_plr_creation_alerts.yaml'
+
+tests:
+  - interval: 1m
+    input_series:
+      # Simulating data from Cluster 1 (crosses the 10% threshold)
+      - series: 'integration_svc_snapshot_created_to_pipelinerun_with_static_env_started_seconds_bucket{le="5", source_cluster="cluster01"}'
+        values: '0+10x5'  # 10 requests took less than 5s
+      - series: 'integration_svc_snapshot_created_to_pipelinerun_with_static_env_started_seconds_bucket{le="+Inf", source_cluster="cluster01"}'
+        values: '0+100x5'  # 100 total occurrences in cluster01
+
+      # Simulating data from Cluster 2 (does not cross the 10% threshold)
+      - series: 'integration_svc_snapshot_created_to_pipelinerun_with_static_env_started_seconds_bucket{le="5", source_cluster="cluster02"}'
+        values: '0+90x5'  # 90 requests took less than 5s
+      - series: 'integration_svc_snapshot_created_to_pipelinerun_with_static_env_started_seconds_bucket{le="+Inf", source_cluster="cluster02"}'
+        values: '0+100x5'  # 100 total occurrences in cluster02
+
+    alert_rule_test:
+      - eval_time: 10m
+        alertname: LatencySnapshotToStaticIntegrationPLRCreation
+        exp_alerts:
+          - exp_labels:
+              severity: warning
+              slo: true
+              source_cluster: cluster01
+            exp_annotations:
+              summary: Latency from an application snapshot created to integration PLRs in static envs created
+              description: >
+                Time from Snapshot created to integration PLRs in static envs created has been over
+                5s for more than 10% of requests during the last 5 minutes on cluster
+                cluster01
+              runbook_url: https://gitlab.cee.redhat.com/rhtap/docs/sop/-/blob/main/integration-service/latency_snapshot_to_integration_test_static.md
+
+# Scenario where both clusters cross the 10% threshold
+# Alert triggered for both clusters
+  - interval: 1m
+    input_series:
+      # Simulating data from Cluster 1 (crosses the 10% threshold)
+      - series: 'integration_svc_snapshot_created_to_pipelinerun_with_static_env_started_seconds_bucket{le="5", source_cluster="cluster01"}'
+        values: '0+5x5'  # 5 requests took less than 5s
+      - series: 'integration_svc_snapshot_created_to_pipelinerun_with_static_env_started_seconds_bucket{le="+Inf", source_cluster="cluster01"}'
+        values: '0+100x5'  # 100 total occurrences in cluster01
+
+      # Simulating data from Cluster 2 (also crosses the 10% threshold)
+      - series: 'integration_svc_snapshot_created_to_pipelinerun_with_static_env_started_seconds_bucket{le="5", source_cluster="cluster02"}'
+        values: '0+5x10'  # 5 requests took less than 5s
+      - series: 'integration_svc_snapshot_created_to_pipelinerun_with_static_env_started_seconds_bucket{le="+Inf", source_cluster="cluster02"}'
+        values: '0+100x5'  # 100 total occurrences in cluster02
```

Comment:
The two lines below are interchangeable: 
```
values: '0+5x10'
values: '0 5 10 15 ... 50'
```
This means that on the first series, you give 11 minutes of measurements to the le=5 series while on the second series you give 6 minutes of measurements to the le=inf series.

I'm not sure this is what you're after given the comments.

See reference here: https://prometheus.io/docs/prometheus/latest/configuration/unit_testing_rules/#series

Reply:
Thank you very much for helpful comments, I messed a lot.